Source/Core/Core/HW/DSPHLE/UCodes/AXWii.cpp

```diff
@@ -436,7 +443,8 @@ void AXWiiUCode::ProcessPBList(u32 pb_addr)
       for (int curr_ms = 0; curr_ms < 3; ++curr_ms)
       {
         ApplyUpdatesForMs(curr_ms, pb, num_updates, updates);
-        ProcessVoice(pb, buffers, spms, ConvertMixerControl(HILO_TO_32(pb.mixer_control)),
+        ProcessVoice(static_cast<HLEAccelerator*>(m_accelerator.get()), pb, buffers, spms,
```

Comment:
Are the casts here necessary? m_accelerator is already an HLEAccelerator* instance under the hood.

Reply:
I think so. `HLEAccelerator` is actually two separate classes; in `AX.cpp` it's `AXGC::HLEAccelerator` and in `AXWii.cpp` it's `AXWii::HLEAccelerator` which are both generated from the same piece of code in `AXVoice.h` via different preprocessor defines. That means that `m_accelerator` must store a pointer to a common base class, but `AcceleratorSetup()` needs the derived class so it can write to the `acc_pb` member.

IMO it's all a bit ugly and I definitely think this could be solved better, so I guess if we wanted we could refactor this further here.